spark/src/main/scala/io/ddf/spark/content/ViewHandler.scala

```diff
@@ -51,10 +56,37 @@ class ViewHandler(mDDF: DDF) extends io.ddf.content.ViewHandler(mDDF) with IHand
     this.get(columns, ViewFormat.withName(format))
   }
 
-  override  def getRandomSampleByNum(numSamples: Int, withReplacement: Boolean,
+  override  def getRandomSampleByNum(numSamples: Long, withReplacement: Boolean,
     seed:
   Int): DDF = {
-    null.asInstanceOf[DDF]
+
+    if (numSamples < 0) {
+      throw new IllegalArgumentException("Number of samples must be larger than or equal to 0")
+    }
+
+    if (!withReplacement){
+      mDDF.getSqlHandler.sql2ddf(s"select * from ${mDDF.getSchema.getTableName} order by rand($seed) limit $numSamples")
+    } else {
+
+      val numRows = mDDF.getNumRows
+      val rddRow: RDD[Row] = mDDF.asInstanceOf[SparkDDF].getRDD(classOf[Row])
+      // We use Spark's API to sample twice what we need, then only pick the first numSamples rows.
+      val sampledRDD = rddRow.sample(true, 2.0 * numSamples/numRows, seed).zipWithIndex().filter(_._2 < numSamples).map(_._1)
+
+      val df: DataFrame = mDDF.getRepresentationHandler.get(classOf[DataFrame]).asInstanceOf[DataFrame]
+      val sqlContext = mDDF.getManager.asInstanceOf[SparkDDFManager].getHiveContext
+      val sampledDF = sqlContext.createDataFrame(sampledRDD, df.schema)
```

Comment:
This might be unnecessary. We can create a DDF directly from an RDD[Row], right?